packages/legend-studio/src/stores/editor-state/element-editor-state/mapping/PureInstanceSetImplementationState.ts

```diff
@@ -271,9 +274,22 @@ export class PureInstanceSetImplementationState extends InstanceSetImplementatio
    * so here we make sure that we reuse existing state and only add new decorated ones
    */
   decorate(): void {
+    const propertyMappingsBeforeDecoration =
+      this.mappingElement.propertyMappings;
```

Comment:
Why don't we make this part of `MappingElementDecorator` instead? Is that possible?